lib/pltr/v2/actions/__tests__/ui.test.js

```diff
@@ -224,3 +253,153 @@ describe('cardDialog', () => {
+    })
+
+    it('should have `null` as initial values for cardId lineId and beatId', () => {
+      expect(bookId).toBeNull()
+      expect(bookNumber).toBeNull()
+    })
+
+    it('should have false as initial value for isOpen', () => {
+      expect(isOpen).toBeFalsy()
+    })
+  })
+
+  describe('given bookDialog actions are dispatched', () => {
+    describe('and setBookDialogOpen is dispatched with no bookId (addBook)', () => {
+      const nextBookNumber = allBookIdsSelector(initialState).length + 1
+      store.dispatch(setBookDialogOpen(null, nextBookNumber))
+      const previousState = store.getState().present
+
+      const bookDialogBookId = bookDialogBookIdSelector(previousState)
+      const bookDialogBookNumber = bookDialogBookNumberSelector(previousState)
+      const isOpen = isBookDialogVisibleSelector(previousState)
+      const initialTotalNumberOfBooks = allBookIdsSelector(previousState).length
+
+      it('should open the bookDialog', () => {
+        expect(isOpen).toBeTruthy()
+      })
+
+      it('should not have a bookId', () => {
+        expect(bookDialogBookId).toBeNull()
+      })
+
+      it('should change the bookNumber equal to the param passed', () => {
+        expect(bookDialogBookNumber).toEqual(nextBookNumber)
+      })
+
+      describe('given addBook is dispatched with completely blank fields', () => {
+        store.dispatch(addBook())
+        const presentState = store.getState().present
+        const allBooks = allBooksSelector(presentState)
+        const totalBooks = allBookIdsSelector(presentState).length
+
+        it('should work as before and save the book without passing any params', () => {
+          expect(totalBooks).toBeGreaterThan(initialTotalNumberOfBooks)
+
+          const book = mapValues(allBooks, (value, key) => {
+            return value.title == '' ? true : undefined
+          })
+          expect(Object.keys(book).length).toBeGreaterThan(0)
+        })
+      })
+
+      describe('given addBook is dispatched with id and attributes', () => {
+        const allBookIds = allBookIdsSelector(previousState)
+        const newBookId = objectId(allBookIds)
+        store.dispatch(addBook(newBookId, { ...exampleBookAttributes }))
+        const presentState = store.getState().present
+        const allBooks = allBooksSelector(presentState)
+        const totalBooks = allBookIdsSelector(presentState).length
+
+        it('should save the book with its attributes', () => {
+          expect(totalBooks).toBeGreaterThan(initialTotalNumberOfBooks)
+
+          const book = mapValues(allBooks, (value, key) => {
+            return value.title == exampleBookAttributes.title &&
+              value.premise == exampleBookAttributes.premise &&
+              value.genre == exampleBookAttributes.genre &&
+              value.theme == exampleBookAttributes.theme
+              ? true
+              : undefined
+          })
+          expect(Object.keys(book).length).toBeGreaterThan(0)
```

Comment:
I don't think this test makes sense(?)
Surely, the length of keys in the book will still be 4?